packages/demo/src/components/Login/index.js

```diff
@@ -1,35 +1,149 @@
 import React from 'react'
-import {connect} from 'cerebral/react'
+import { connect } from 'cerebral/react'
 import translations from '../../computed/translations'
+import LangSelector from '../LangSelector'
 
 export default connect(
   {
-    t: translations
+    t: translations,
+    $email: 'user.signIn.$email',
+    $password: 'user.signIn.$password',
+    $validationsErrors: 'user.signIn.$validationsErrors',
+    $error: 'user.signIn.$error'
   },
-  function Login ({t}) {
+  {
+    fieldChanged: 'user.fieldChanged',
+    signInClicked: 'user.signInClicked',
+    createUserClicked: 'user.createUserClicked'
+  },
+  function Login ({
+    t,
+    // state
+    $email,
+    $password,
+    $validationsErrors,
+    $error,
+    // signals
+    fieldChanged,
+    signInClicked,
+    createUserClicked
+  }) {
+    function getField ({
+      field,
+      icon,
+      value,
+      placeholder,
+      fieldType,
+      message
+    }) {
+      return (
+        <p className='control has-icon'>
+          <input className='input'
+            type={fieldType}
+            placeholder={placeholder}
+            value={value}
+            onChange={e => fieldChanged({
+              field,
+              value: e.target.value
+            })}
+          />
+          {message && (
+            <i className='fa fa-warning' />
+          )}
+          {message && (
+            <span className='help is-danger'>
+              {message}
+            </span>
+          )}
+          {!message && (
+            <i className={icon} />
+          )}
+        </p>
+      )
+    }
+
+    function getEmail () {
+      let hasValidationError = $validationsErrors && $validationsErrors.EMAIL_EMPTY
+      let hasError = $error
+      let message = ''
+      if (hasValidationError) {
+        message = t.loginValidationEmailRequired
+      } else if (hasError) {
+        let translation = t.loginErrors[ $error.code ]
+        message = translation || $error.message
+      }
+      return getField({
+        field: 'user.signIn.$email',
+        icon: 'fa fa-user',
+        value: $email,
+        placeholder: t.loginEmailPlaceholder,
+        fieldType: 'email',
+        message: message
+      })
```

Comment:
With a separate Login/Input.js (see comment above), this would become:

```js
return <Input field='...' ... />
```

Makes code easier to understand I think...